File: HMCalcWSIZ.Core/Domain/Operation.cs
using System;
using HMCalcWSIZ.Core.DTO;

namespace HMCalcWSIZ.Core.Domain
{
    public class Operation
    {
        public int Id { get; set; }

        public DateTime OperationDate { get; set; }

        public DateTime CreationDate { get; set; }

        public decimal Amount { get; set; }

        public bool IsIncome { get; set; }

        public string Description { get; set; }

        public bool IsCycle { get; set; }

        public int? CycleId { get; set; }

        public virtual Operation Cycle { get; set; }

        public string UserId { get; set; }

        public virtual AppUser User { get; set; }

        public OperationDTO ToDto()
        {
            return new OperationDTO
            {
                Id = Id,
                OperationDate = OperationDate,
                Amount = Amount,
                CreateDate = CreationDate,
                Description = Description,
                UserId = UserId,
                IsIncome = IsIncome,
                IsCycle = IsCycle,
            };
        }
    }
}

File: HMCalcWSIZ/ClientApp/src/app/models/operation.ts
export class Operation {
    _id: number;
    operationDate: Date;
    isIncome: boolean;
    amount: number;
    description: string;
    isCycle: boolean;
    cycleId: number;

    constructor(id: number) {
        this._id = id;
        this.operationDate = new Date();
        this.operationDate.setUTCHours(12);
        this.isIncome = true;
        this.amount = 1;
        this.description = '';
        this.isCycle = false;
        this.cycleId = null;
    }

    static createArray(res): Operation[] {

        let result: Operation[];
        result = res;

        result.forEach((value, index) => {
            Operation.normalize(value, res[index].isIncome, res[index].id);
        });
        return result;
    }

    static normalize(operation: Operation, type, id: number) {
        operation.isIncome = (type === '1') || (type === true);
        operation._id = id;
    }
}
